challenges-14-plus.yml

```diff
@@ -0,0 +1,472 @@
+        <h1>Welcome to our Pizza Section!</h1>
+
+        <p>You need to be a pizza lover before you proceed!</p>
+
+        <button
+            class="btn btn-primary"
+            onclick="javascript:document.getElementById('my_great_modal').style.display='none';"
+        >
+            Proceed
+        </button>
+    </div>
+    ```
+    
+    This works as expected, but I'm a bit nervous about leaving too much time
+    on our users to decide. They should be quick! So I've added a timeout on
+    the main JavaScript of the page to force users *away* after a certain
+    amount of time.
+    
+    ```javascript
+    window.setTimeout(() => {
+        window.location.href = '/';
+    }, 10000);
+    ```
+    
+    How can we use Turbo Drive (instead of just `window.location.href=`)
+    to perform a page visit without doing a full refresh?
+  explanation: |
+    By using the syntax `import * as Turbo from '@hotwired/turbo'` you will be importing all
+    public functions from the library into an object called `Turbo`. This enables
+    you to use some Turbo Drive specific actions in your scripts, like `Turbo.visit`.
+    
+    > You can also use destructuring to import only the functions you need. In
+    > this case, this would be: `import { visit } from '@hotwired/turbo'`.
+    
+    > Turbo is also set automatically as a global variable once you import it anywhere in your
+    > page, so you can also use Turbo from inline scripts!
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          ```javascript
+          import Turbo from '@hotwired/turbo`;
+
+          window.setTimeout(() => {
+              Turbo.visit('/');
+          }, 10000);
+          ```
+        b: |
+          ```javascript
+          import * as Turbo from '@hotwired/turbo`;
+
+          window.setTimeout(() => {
+              Turbo.location.href = '/';
+          }, 10000);
+          ```
+        c: |
+          ```javascript
+          import * as Turbo from '@hotwired/turbo`;
+
+          window.setTimeout(() => {
+              Turbo.visit = '/';
+          }, 10000);
+          ```
+        d: |
+          ```javascript
+          import * as Turbo from '@hotwired/turbo`;
+          
+          window.setTimeout(() => {
+              Turbo.visit('/');
+          }, 10000);
+          ```
+      correct: d
+      
+transitions:
+  question: |
+    I just came back from a meeting with our designer, and she has an *amazing* concept
+    for a new transition that we need to add to our page loads between clicks! It looks like
+    this in the CSS main file:
+    
+    ```css
+    body {
+        background-color: #FFF
+        transition: background-color 0.2s;
+    }
+    
+    body.turbo-loading {
+        background-color: #F00;
+    }
+    ```
+  
+    I think this is going to look *awesome* (and won't render too many of our viewers blind)!
+    
+    So, how can we implement these transitions using Turbo Drive event listeners?
```

Comment:
Hmm. This is a *really* technical question. I'm not sure I like it. Doing this stuff is hard, and the important thing is that the user understands the general concepts and then can (more or less) copy our implementation.

Would it be possible to give them an implementation with a mistake, and ask them what's wrong with it? Or, show an implementation with a mistake. Then a "coworker" shows us the solution. Then we ask: why did this solve it?

For example, maybe we show the solution (which is `b` below), but without `requestAnimationFrame()` (we just remove the class right inside `turbo:render`). Then our co-worker tells us the fix. And we ask: why did this fix things? It's a bit of a simpler way to have them think critically about a piece of it - vs going back to the video to compare the code in the video with the code in each answer.